param/__init__.py

```diff
@@ -36,10 +36,13 @@
 # only two required files.
 try:
     from .version import Version
-    __version__ = Version(release=(1,5,1), fpath=__file__,
-                          commit="$Format:%h$", reponame="param")
+    _version = Version(release=None, fpath=__file__,
```

Comment:
Will autover accept (1,5,1) and do something reasonable with it, so that people who have been using the old param.version.Version won't have problems?

Reply:
I *think* so as ``autover.Version`` is based on the old ``param.Version``. The issues are:

1) Supporting the old behavior properly makes things more complicated.
2) This old approach is not properly tested.

I'm pretty sure it will be accepted but I can't guarantee it will do the right thing! I think the original unit tests were moved to autover but I would need to check how many remain.

Reply:
I should add that we are the only people who used ``param.Version`` as far as I know (unlikely autover it wasn't a separate project and it wasn't advertised).

Reply:
One option would be to explicitly *disallow* these tuples making sure that projects depending on the next param release have to update to the new approach.

Reply:
We haven't advertised it recently, but it has always been available for use, the code at the top of the old Version file explained exactly how to use it in external projects, and though it's been years now, I definitely recommended it in various talks to audiences I have now forgotten.

Reply:
I'm not sure what to do about this. On one hand I *think* this is mostly backwards compatible (other than the deprecated ``commit_count`` argument that wasn't often used) but I am not convinced that it is worth the effort to promise backwards compatibility add new tests etc. Even if it an explicit release tuple is fine, the actual version string format has changed and is not PEP440 compliant.

I think the honest thing to do here would be to issue a warning if ``release`` is not ``None`` or if deprecated arguments like ``commit_count`` are used. That way if any projects are using ``param.Version`` that we aren't in control of *also* upgrade their param version, they get a warning so they know that they should at least check that ``Version`` still works appropriately for their projects (and to prompt them to know that there is a new recommended workflow that is easier, purely based on tags).

Reply:
That sounds fine to me.

Reply:
Shouldn't release be None by default, then?

Reply:
It is: https://github.com/ioam/param/pull/214/files#diff-0187a55516aa9995e9033ad5f0d98b33R170

Reply:
The way it is specified in the line above is explicit but not strictly necessary. Explicit is good though!

Reply:
But since this is code that gets copied into every new project, I'd rather have it as short as possible here.

Reply:
Fixed in 0954adf. The version declaration shouldn't be copied verbatim between projects but I agree it is worth making this short as it is the reference example!